### PR TITLE
BookB-Pset_GeoScienceFeatureTypeSoilLikeMaterial

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_GeoScienceFeatureTypeSoilLikeMaterial

## Description: 

Properties for soil like material

| Property | Datatype | Description |
|----------|----------|-------------|
|Lithology|IfcIdentifier|Defines all natural materials resulting from the combination of minerals, particles or biogenic elements.|
|WaterContent|IfcPositiveRatioMeasure|Ratio of the mass of water to the mass of solids.|
|VoidRatio|IfcPositiveRatioMeasure|Ratio of the volume of void to the volume of solid.|
|WetBulkDensity|IfcMassDensityMeasure|Ratio of the total mass to the unit total volume (material at its natural moisture content).|
|DryBulkDensity|IfcMassDensityMeasure|Ratio of the mass of solid to the unit total volume (dry material).|
|YoungModulus|According to existing definition|According to existing definition|
|MenardLimitPressure|IfcPressureMeasure|Limit pressure, resulting from interpretation of Menard's pressuremeter test.|
|CPTconeResistance|IfcPressureMeasure|Cone penetration resistance measured during the CPT test. |
|SPTBlowCount|IfcCountMeasure|Number of blows N required to drive the sampler for a test drive of 300 mm following the seating drive.|
|PlasticityIndex|IfcPositiveRatioMeasure|The plasticity index Ip is the numerical difference between the liquid limit and plastic limit of a fine soil.  IP = wL - wP|
|ConsistencyIndex|IfcPositiveRatioMeasure|The consistency index (Ic) indicates the consistency (firmness) of a soil. It is the numerical difference between the liquid limit and the natural water content expressed as a percentage ratio of the plasticity index. Ic = (LL-W)/Ip|
|PeakCohesion|IfcPressureMeasure|Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the shear resistance of soils that is independent from the normal effective stress. The peak value correspond to the maximum shear measured during the test.|
|ResidualCohesion|IfcPressureMeasure|Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the shear resistance of soils that is independent from the normal effective stress. The residual value correspond to the shear strength measured at large strain values during the test.|
|PeakFrictionAngle|IfcPlaneAngleMeasure|Soil effective stress friction angle is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the friction shear resistance of soils with the normal effective stress. The peak value correspond to the maximum shear strength measured during the test.|
|ResidualFrictionAngle|IfcPlaneAngleMeasure|Soil effective stress friction angle is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the friction shear resistance of soils with the normal effective stress. The residual value correspond to the shear strength measured at large strain values during the test.|
|UndrainedShearStrength|IfcPressureMeasure|Shear strength of a soil under undrained condition or constant volume conditions. It is relevant for conditions where the rate of loading is very much greater than the rate at which pore water pressure - generated due to the action of shearing of the soil - dissipates.  Its calculation depends on the laboratory test performed.|
|PoissonRatio|According to existing definition|According to existing definition|
|CompressionIndex|IfcReal|Compression index is determined with the oedometer tests, and used to compute settlements.|
|SwellingIndex|IfcReal|Swelling index is determined with the oedometer test, and used to compute swelling.|
|OCR|IfcReal|Ration between preconsolidation pressure and overburden pressure.|
|K0|IfcPositiveRatioMeasure|The coefficient of lateral earth pressure, K, is defined as the ratio of the horizontal effective stress, σ’h, to the vertical effective stress, σ’v. For a level ground deposit with zero lateral strain in the soil, the "at-rest" coefficient of lateral earth pressure, K0 is obtained.|
|RelativeDensity|IfcPositiveRatioMeasure|Relative density or density index is the ratio of the difference between the void ratios of a cohesionless soil in its loosest state and existing natural state to the difference between its void ratio in the loosest and densest states. (emax - e) /(emax-emin).|

### CODE REVIEW

File: IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu" Name="CompressionIndex" UniqueId="6e329162-f3d9-40c2-a228-bd24ea4773f8" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml]: ConsistencyIndex is a ratio measure. Why is this one `IfcReal`?

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Emodulus_10zwRze1ElBZzkStdNfro" Name="Emodulus" UniqueId="40f7a6ff-f680-4ebc-b8fd-b9cde75e9d72" PrimaryDataType="IfcModulusOfElasticityMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Reuse existing "YoungModulus".

File: IFC4x3/Properties/l/Lithology_3Ib9wYZlH8rvMKekRGTYia/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Lithology_3Ib9wYZlH8rvMKekRGTYia" Name="Lithology" UniqueId="d2949ea2-8ef4-48d7-9594-a2e6d0762b24" PrimaryDataType="IfcIdentifier" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/l/Lithology_3Ib9wYZlH8rvMKekRGTYia/DocProperty.xml]: Shall be reused.

File: IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Ration between preconsolidation pressure and overburden pressure.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/Documentation.md]:
```suggestion
Ratio between pre-consolidation pressure and overburden pressure.
```

File: IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/DocProperty.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="OCR_3J8URKaDb0hfGVNxhjZK2p" Name="OCR" UniqueId="d321e6d4-90d9-40ae-941f-5fbaed8d40b3" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/DocProperty.xml]: Propose one of the ratio measure data types.

File: IFC4x3/Properties/p/PeakCohesion_1TVc0spc13oOyuIwsuBSiU/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the shear resistance of soils that is independent from the normal effective stress. The peak value correspond to the maximum shear measured during the test.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PeakCohesion_1TVc0spc13oOyuIwsuBSiU/Documentation.md]:
```suggestion
Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and is used to describe the shear resistance of soils that is independent from the normal effective stress. The peak value correspond to the maximum shear measured during the test.
```

File: IFC4x3/Properties/p/PeakFrictionAngle_0z1h4Y4EvBmefF2zOURpEQ/Documentation.md [hunk 7]
```diff
@@ -0,0 +1 @@
+Soil effective stress friction angle is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the friction shear resistance of soils with the normal effective stress. The peak value correspond to the maximum shear strength measured during the test.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PeakFrictionAngle_0z1h4Y4EvBmefF2zOURpEQ/Documentation.md]:
```suggestion
Soil effective stress friction angle is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and is used to describe the friction shear resistance of soils with the normal effective stress. The peak value correspond to the maximum shear strength measured during the test.
```

File: IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md [hunk 8]
```diff
@@ -0,0 +1 @@
+The plasticity index Ip is the numerical difference between the liquid limit and plastic limit of a fine soil.  IP = wL ? wP.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md]: Not sure.
```suggestion
The plasticity index Ip is the numerical difference between the liquid limit and plastic limit of a fine soil.  IP = wL / wP.
```

File: IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/DocProperty.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O" Name="PlasticityIndex" UniqueId="956df50f-8aee-4758-b2cc-f8e9dca2e158" PrimaryDataType="IfcPositiveRatioMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/DocProperty.xml:2]: Shall be reused.

File: IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb" Name="PoissonRatio" UniqueId="0750cf69-0deb-4d99-84f3-2659d120d965" PrimaryDataType="IfcPositiveRatioMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml]: Reuse existing.

File: IFC4x3/Properties/r/ResidualCohesion_21T5RytGz4yvSfcQD$D7p2/Documentation.md [hunk 11]
```diff
@@ -0,0 +1 @@
+Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and it is used to describe the shear resistance of soils that is independent from the normal effective stress. The residual value correspond to the shear strength measured at large strain values during the test.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/ResidualCohesion_21T5RytGz4yvSfcQD$D7p2/Documentation.md]:
```suggestion
Cohesion is a shear strength parameter of soils. Its definition is derived from the Mohr-Coulomb failure criterion and is used to describe the shear resistance of soils that is independent from the normal effective stress. The residual value correspond to the shear strength measured at large strain values during the test.
```

File: IFC4x3/Properties/s/SPTBlowCount_35XQNkKeH4W8XkaRfglEJO/Documentation.md [hunk 12]
```diff
@@ -0,0 +1 @@
+Number of blows N required to drive the sampler for a test drive of 300 mm following the seating drive.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SPTBlowCount_35XQNkKeH4W8XkaRfglEJO/Documentation.md]:
```suggestion
Number of blows required to drive the sampler for a test drive of 300 mm following the seating drive.
```

File: IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SwellingIndex_2gcMqEfT7o9RCMSQREDB" Name="SwellingIndex" UniqueId="aa9bf5b4-3a97-47c8-96cc-59c69b38dfcb" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml]: Ratio measure or IfcReal?

File: IFC4x3/Properties/v/VoidRatio_3iyX_eZCr9MgXBX7IpP$HN/DocProperty.xml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VoidRatio_3iyX_eZCr9MgXBX7IpPHN" Name="VoidRatio" UniqueId="ecf21fa8-8ccd-495a-a84b-8474b367f457" PrimaryDataType="IfcPositiveRatioMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/v/VoidRatio_3iyX_eZCr9MgXBX7IpP$HN/DocProperty.xml]: Shall be reused.

File: IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml]: IfcReal in Excel spec, but I had the samew question. @j-weil

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Same comment as many other places. Hopefully with one generic answer.

File: IFC4x3/Properties/l/Lithology_3Ib9wYZlH8rvMKekRGTYia/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/l/Lithology_3Ib9wYZlH8rvMKekRGTYia/DocProperty.xml]: Yes it is reused as far as I can see.

File: IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml (hunk 10 above)
Comment (larswik) [reply IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml]: See also [BookB-Pset_GeoScienceFeatureTypeIntactRock](https://github.com/bSI-InfraRoom/IFC-Specification/pull/627#top).
There is one "PoisonsRatio" used in Pset_SolidStratumCapacity which feels named wrongly but with a generic enough definition: "Ratio of transverse contraction strain to longitudinal extension strain in the direction of stretching force.". Then there is one "PoissonRatio" with the definition "A measure of the lateral deformations in the elastic range." which feels a bit vague to me. Can we generalize all these into one?
The one used here has the definition "Ratio between radial/horizontal strain and axial strain of a (geo)material when compressed."

File: IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml (hunk 13 above)
Comment (larswik) [reply IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml]: IfcReal in excel spec. But let's ask @j-weil.

File: IFC4x3/Properties/v/VoidRatio_3iyX_eZCr9MgXBX7IpP$HN/DocProperty.xml (hunk 14 above)
Comment (larswik) [reply IFC4x3/Properties/v/VoidRatio_3iyX_eZCr9MgXBX7IpP$HN/DocProperty.xml]: Yes.

File: IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/DocProperty.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/o/OCR_3J8URKaDb0hfGVNxhjZK2p/DocProperty.xml]: Set to IfcPositiveRatioMeasure in [6ccafb6](https://github.com/bSI-InfraRoom/IFC-Specification/pull/630/commits/6ccafb63b1f7490c8973e576f1478cabb2ddf70f).

File: IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md]: Should be IP = wL - wP. Also re-used.

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Fixed in [e263b67](https://github.com/bSI-InfraRoom/IFC-Specification/pull/630/commits/e263b67252fc5988e33bf8c5c130de5ca8fc56a8)

File: IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml (hunk 10 above)
Comment (larswik) [reply IFC4x3/Properties/p/PoissonRatio_07KCzf3UjDcOJp9bdH8Dbb/DocProperty.xml]: Fixed in [b5a0c83](https://github.com/bSI-InfraRoom/IFC-Specification/pull/630/commits/b5a0c83eb99dfdf079584512ed47156833ffba45)

File: IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/p/PlasticityIndex_2LRVKFYkv7MBBC_EdSek5O/Documentation.md]: Fixed in [1315b36](https://github.com/bSI-InfraRoom/IFC-Specification/pull/630/commits/1315b36099fc1e48e7b012240da0b1894a803ab2).

File: IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/c/CompressionIndex_1kCf5Yyzb0mg8elIJgHtFu/DocProperty.xml]: Changed to IfcRatioMeasure (IfcPositiveRatioMeasure?)

File: IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml (hunk 13 above)
Comment (larswik) [reply IFC4x3/Properties/s/SwellingIndex_2gc$MqEfT7o9RCMSQRED$B/DocProperty.xml]: IfcRatioMeasure.

**Pull request merged**